### PR TITLE
expression: support builtin function json_contains

### PR DESCRIPTION
**Pull request by mccxj**
### What problem does this PR solve? 

https://github.com/pingcap/tidb/issues/7299

### What is changed and how it works?

add builtin function json_contains according to https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-contains

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
add exported method PathExpression#ContainsAnyAsterisk()

Related changes

waiting for https://github.com/pingcap/tipb/pull/99

### CODE REVIEW

**Comment by jackysp** [comment]
Thanks for your contribution, @mccxj !

**Comment by mccxj** [comment]
@jackysp PTAL

**Comment by jackysp** [comment]
PTAL @hicqu @coocood

**Comment by jackysp** [comment]
PTAL @hicqu @coocood

**Comment by coocood** [comment]
LGTM

File: expression/builtin_json.go [hunk 1]
```diff
@@ -548,3 +551,66 @@ func jsonModify(ctx sessionctx.Context, args []Expression, row chunk.Row, mt jso
...
+	obj, isNull, err := b.args[0].EvalJSON(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+	target, isNull, err := b.args[1].EvalJSON(b.ctx, row)
+	if isNull || err != nil {
+		return res, isNull, errors.Trace(err)
+	}
+	var pathExpr json.PathExpression
+	if len(b.args) == 3 {
```
Comment (zz-jason) [inline expression/builtin_json.go]: It better to write another function signature to handle the scenario that the third argument is provided.

**Pull request merged**